### DISCUSSION TITLE
Proposal for Version Alignment

### DISCUSSION THREAD

**Discussion by benj0c** [MurhafSousli/ngx-sharebuttons#695] [Ideas]
I would like to propose a new versioning system for this library. The idea is to synchronize the library’s versioning with Angular’s major versions. For instance, version 16 would correspond with Angular version 16, and so on.

This approach offers several key advantages:

- Clarity: It provides an intuitive understanding of compatibility between ngx-sharebuttons and Angular. Users can easily identify which library version works with their current Angular version.
- Simplified Maintenance: Aligning the versions minimizes confusion and reduces the time spent cross-referencing versions, especially during upgrades.
- User Confidence: By signaling clear support for each major Angular release, we foster trust and encourage users to adopt new versions with confidence, knowing they are fully supported.

This change can streamline development and usage for both contributors and users of the library. I believe this would greatly enhance the overall developer experience and contribute to a more consistent update cycle.

What are your thoughts on this approach?

**Comment by MurhafSousli** [comment]
I agree with key features you mentioned, however, this could potentially introduce some risks.

Lets say both Angular and the library version are v18. and we needed to introduce a breaking changes or a complete re-write. then we won't be able to bump the major version.

Angular bumps their major version regularly every 6 months regardless if they introduced breaking changes or not. and most often the library will work fine with the newer Angular versions.

For a library of this size, it adds more effort to keep up to Angular versions. which I prefer to avoid.
